File: Chapter 03/JS/section_3_3_5_dip/after/winger.js
/**
 * DIP SOLUTION: The "Plug-and-Play" Proof.
 * * ARCHITECTURE NOTE: This class proves the system is now open for extension. 
 * We created a Winger and can immediately include it in the game plan. 
 * * Most importantly: We did not have to touch a single line of code in 
 * coach.js to add this new player type!
 */
class Winger {
    performAction() {
        console.log("  [Action] Winger is running down the sideline.");
    }
}

module.exports = Winger;

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_5_dip/after/Winger.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_5_dip.after;

/**
 * DIP SOLUTION: The "Plug-and-Play" Proof.
 * * ARCHITECTURE NOTE: Because the Coach depends on an abstraction, we were 
 * able to create this brand new Winger class and instantly inject it into 
 * the Coach's game plan without altering a single line of the Coach class! 
 * * This is the ultimate goal of software architecture: adding new features 
 * by adding new code, rather than modifying existing code.
 */
public class Winger implements Player {
    public void performAction() {
        System.out.println("  [Action] Winger is running down the sideline.");
    }
}